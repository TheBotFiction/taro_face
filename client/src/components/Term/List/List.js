/**
 * @flow
 */
import React, { PureComponent } from 'react'
import { Link } from 'react-router-dom'
import type { TermListItem } from 'types'
import { Paper, Table, TableBody, TableCell, TableHead, TableRow } from '@material-ui/core'

type Props = {
  terms: TermListItem[]
}

class TermList extends PureComponent<Props, *> {
  render() {
    const termItems = this.props.terms.map((term, i) => {
      return (
        <TableRow key={i}>
          <TableCell style={{ width: '20%' }}>
            <Link to={`/terms/${term.id}`}>
              {term.term}
            </Link>
          </TableCell>
          <TableCell>
            <p><strong>Meaning:</strong> {term.meaning}</p>
            <p>
              <strong>Sample:</strong><br/>
              <ul>
                {
                  term.samplePhrases.map((phrase, i) =>
                    <li key={i}>{phrase.phrase}</li>
                  )
                }
              </ul>
            </p>
          </TableCell>
        </TableRow>
      )
    })
    return (
      <Paper>
        <Table>
          <TableHead>
            <TableRow>
              <TableCell>Term</TableCell>
              <TableCell>Details</TableCell>
            </TableRow>
          </TableHead>
          <TableBody>
            {termItems}
          </TableBody>
        </Table>
      </Paper>
    )
  }
}

export default TermList
