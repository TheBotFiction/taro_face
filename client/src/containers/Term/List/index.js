import TermListContainer from './List'

export default TermListContainer
