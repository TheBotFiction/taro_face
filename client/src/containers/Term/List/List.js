/**
 * @flow
 */
import React, { PureComponent } from 'react'
import type { Node } from 'react'
import type { TermListItem } from 'types'
import { Query } from 'react-apollo'
import gql from 'graphql-tag'
import { TermList } from 'components/Term'

type Props = { }

type State = {}

type QueryResult = {
  loading: boolean,
  data: {
    terms: TermListItem[]
  }
}

const INDEX_TERM_QUERY = gql`
  query indexTerm {
    terms {
      id
      term
      meaning
      samplePhrases {
        id
        phrase
      }
    }
  }
`
class TermListContainer extends PureComponent<Props, State> {
  render () {
    return (
      <Query query={INDEX_TERM_QUERY}>
        {({loading, data: { terms }}: QueryResult): Node => {
          if (loading) return <h2>Loading ...</h2>
          return (
            <TermList terms={terms} />
          )
        }}
      </Query>
    )
  }
}

export default TermListContainer
