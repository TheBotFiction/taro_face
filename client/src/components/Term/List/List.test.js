import React from 'react'
import TestRenderer from 'react-test-renderer'
import { shallow } from 'enzyme'

import List from './List'

describe('component Term/List', () => {
  it('renders without crash', () => {
    shallow(
      <List terms={[]} />
    )
  })

  it('renders as expected', () => {
    const component = TestRenderer.create(
      <List terms={[]} />
    )
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  })
})
