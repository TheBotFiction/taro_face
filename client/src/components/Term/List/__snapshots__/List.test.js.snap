// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component Term/List renders as expected 1`] = `
<div
  className="MuiPaper-root-1 MuiPaper-elevation2-5 MuiPaper-rounded-2"
>
  <table
    className="MuiTable-root-28"
  >
    <thead
      className="MuiTableHead-root-29"
    >
      <tr
        className="MuiTableRow-root-30 MuiTableRow-head-33"
      >
        <th
          aria-sort={null}
          className="MuiTableCell-root-35 MuiTableCell-head-36"
          scope="col"
        >
          Term
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root-35 MuiTableCell-head-36"
          scope="col"
        >
          Details
        </th>
      </tr>
    </thead>
    <tbody
      className="MuiTableBody-root-43"
    />
  </table>
</div>
`;
